smartsim/settings/base.py

```diff
@@ -511,24 +520,20 @@ def _build_exe_args(exe_args: t.Optional[t.Union[str, t.List[str]]]) -> t.List[s
                 return exe_args.split()
             if isinstance(exe_args, list):
                 exe_args = copy.deepcopy(exe_args)
-                plain_type = all([isinstance(arg, (str)) for arg in exe_args])
+                plain_type = all(isinstance(arg, (str)) for arg in exe_args)
                 if not plain_type:
                     nested_type = all(
-                        [
-                            all([isinstance(arg, (str)) for arg in exe_args_list])
-                            for exe_args_list in exe_args
-                        ]
+                        all(isinstance(arg, (str)) for arg in exe_args_list)
```

Comment:
Might want to double-check that this is equivalent

Reply:
If I remember correctly, there were two rounds of collapsing all() calls here, so I wanted to just double-check that it was equivalent

Reply:
I added a couple tests where I could easily exercise the code for creating arguments.

For reference, here's the snippet of nested `all`s

```
if isinstance(exe_args, list):
  exe_args = copy.deepcopy(exe_args)
  plain_type = all(isinstance(arg, (str)) for arg in exe_args)
  if not plain_type:
      nested_type = all(
          all(isinstance(arg, (str)) for arg in exe_args_list)
          for exe_args_list in exe_args
      )
      if not nested_type:
          raise TypeError(
              "Executable arguments were not list of str or str"
          )
      return exe_args
  return exe_args
```
I added tests that exercise all the possibilities not covered in existing tests:

- exe_args is a list instead of a string.
    - `exe_args = [script, "--time", "3"]`
- exe_args is a list containing a non-string.
    - `exe_args = [script, "--time", 3]`
- exe_args is a list-of-lists containing a non-string (exercise nested alls):
    - `exe_args = [[script, "--time", 3]]`
- exe_args is a list-of-lists containing only strings (exercise nested alls):
    - `exe_args = [[script, "--time", "3"]]`

I also added a ticket to review inconsistent behaviors between adding exe_args in the run settings constructor & via `add_exe_args` method.